Dockerfile

```diff
@@ -13,8 +15,10 @@ RUN go mod download
 COPY main.go main.go
 COPY pkg/ pkg/
 
+ARG TARGETARCH
+
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH="${TARGETARCH}" go build -a -o manager main.go
```

Comment:
Can we also add the `ARG TARGETOS` and use it for setting `GOOS`?"

Reply:
`TARGETOS` does not make much sense, as the outcome we want here is always going to be linux.

Reply:
Why? I believe it's better to use the `TARGETOS` value supplied by `buildx` instead of hardcoding it to `linux`, as this Dockerfile wouldn't work for other OS. We can set it like `GOOS=${TARGETOS:-linux}`.

Reply:
The base image used for the final container is RHEL, so the target is not going to change.
Should it change in the future this can be changed to a variable, but right now it will always be the same value.

Reply:
Yes, that makes sense. Thanks!